src/core/rewriter.sk

```diff
@@ -315,6 +318,12 @@ class GLSLX.Rewriter {
       }
 
       case .IF {
+        if node.ifTrue == null {
```

Comment:
Yes. I'd like to take this approach and make sure that if statement ASTs always represent valid syntax. If you look inside parseStatement() you can see that the empty statement ";" is represented by an empty block "{}" instead of null.